crates/mun_abi/src/autogen_impl.rs

```diff
@@ -146,6 +146,22 @@ impl StructInfo {
             unsafe { slice::from_raw_parts(self.field_sizes, self.num_fields as usize) }
         }
     }
+
+    /// Returns the index of the field matching the specified `field_name`.
+    pub fn find_field_index(struct_info: &StructInfo, field_name: &str) -> Result<usize, String> {
```

Comment:
I think at some point we should figure out a way to do some sort of hash lookup. Because this might become slow.